r/R/dplyr-funcs-datetime.R

```diff
@@ -115,6 +115,23 @@ register_bindings_datetime <- function() {
       return(Expression$create("strftime", x, options = list(format = format, locale = locale)))
     }
 
+    if (call_binding("is.integer", x)) {
+      if (inherits(x, "Expression")) {
+        call_binding(
+          "if_else",
+          call_binding_agg("all", call_binding("between", x, 1, 12))$data,
```

Comment:
I suspect that the `$data` isn't doing what you expect here. That's actually getting the _input_ to the `all` binding (not the result).

We also should be careful that we don't want to go through the data more than once — is there a way we can construct a series of binding calls like this that get us to the answer (and then error if they run into a value that won't work?)